### PR TITLE
Delete `~/.npm/_cacache/tmp` before caching the folder

### PR DESCRIPTION
**Pull request by mokagio**
Follow up to @geriux and @dcalhoun comments on https://github.com/wordpress-mobile/gutenberg-mobile/pull/6707 to:

- Prevent `tmp` files name clash
- Address a copy-pasted typo

Testing: Ensure CI is green (o, if red, that the failure were legit test failures unrelated with caching)

---

PR submission checklist:

- [x] I have considered adding unit tests where possible. — This is a tooling PR.
- [x] I have considered if this change warrants user-facing release notes [more info](https://github.com/wordpress-mobile/gutenberg-mobile/blob/trunk/Release-notes.md) and have added them to [RELEASE-NOTES.txt](https://github.com/wordpress-mobile/gutenberg-mobile/blob/trunk/RELEASE-NOTES.txt) if necessary. — Not necessary.

### CODE REVIEW

File: .buildkite/commands/install-node-dependencies.sh [hunk 1]
```diff
@@ -34,13 +34,19 @@ echo "--- :npm: Save cache if necessary"
 # Notice that we don't cache the local node_modules.
 # Those get regenerated by npm ci as per Node reccomendations.
 # What we are caching is the root npm folder, which stores pacakge downloads so they are available if the package.json resolution demands them.
+#
+# npm stores temporary files in ~/.npm that we don't want to extract because they might run into naming conflicts.
+# So, before archiving it, we remove those tmp files.
+#
+# Example: https://buildkite.com/automattic/gutenberg-mobile/builds/8857#018e37eb-7afc-4280-b736-cba76f02f1a3/524
+rm -rf "$HOME/.npm/_cacache/tmp"
```
Comment (mokagio) [inline .buildkite/commands/install-node-dependencies.sh:42]: I looked for docs about the role of the `tmp` folder but came out short.

Visited:

- https://docs.npmjs.com/cli/v9/configuring-npm/folders#cache
- https://docs.npmjs.com/cli/v9/commands/npm-cache
- https://www.npmjs.com/package/cacache

I still think it's safe to remove the `tmp` folder.  As the name suggests, it should only contain temporary files.

Comment (mokagio) [reply .buildkite/commands/install-node-dependencies.sh:42]: I also looked into my local `pnpm` cache for an equivalent, but couldn't find it.

I assume if something with the possibility of name clashing exists there, it'll reveal itself at some point in CI, like the `~/.npm` one did.

Comment (derekblank) [reply .buildkite/commands/install-node-dependencies.sh:42]: > I still think it's safe to remove the tmp folder. As the name suggests, it should only contain temporary files.

I would agree. The [cacache docs on tmp](https://github.com/npm/cacache?tab=readme-ov-file#--tmpwithtmpcache-opts-cb---promise) suggest that the tmp directory may even be designed to delete itself, so it seems safe to programatically do so. But, collision possibilities would reveal themselves in CI, as you said.

**Review by derekblank — APPROVED** [review]
LGTM. 🚀 

@geriux / @dcalhoun Feel free to include any thoughts as well.

**Review by geriux — APPROVED** [review]
LGTM! Since we haven't updated the `package-lock.json` we might see this issue until the stored cached is replaced. I'll merge this so we can include it before today's release and hopefully it'll be fixed 🚀 

Thank you Gio!

**Pull request merged**